Source/ProjectRimFactory/Common/PRFGameComponent.cs

```diff
@@ -80,14 +93,49 @@ public static void RegisterAssemblerQueue(IAssemblerQueue queue)
             }
             return false;
         }
-        /*
-        // A quick way to test all the scupltures available, if need be.       
-        public override void LoadedGame() {
-            base.LoadedGame();
-            foreach (var t in Current.Game.CurrentMap.spawnedThings.Where(x => x is Building_Art)) {
-                if (!Current.Game.GetComponent<PRFGameComponent>().TryAddSpecialSculpture(t)) return;
-                Log.Warning("---------added test special scuplture: " + t + " at " + t.Position);
+
+        public override void LoadedGame()
+        {
+            //Remove all entrys where the Building is not where it says it is
+            //I would prefer to call a AssemblerQueue.Clear but LoadedGame() gets called after all buildings have spawn. therfor this is not possible & i need another way to remove invalid / irelevant entrys
+            for (int i = AssemblerQueue.Count - 1; i >= 0; i--)
+            {
+                if (!AssemblerQueue[i].map.thingGrid.ThingsAt( (AssemblerQueue[i] as Building).Position).Any(t => t.ToString() == AssemblerQueue[i].ToString()))
```

Comment:
`t.ToString() == AssemblerQueue[i].ToString()`
there has to be a better way _(i have this in multiple spots)_

Reply:
Actually....`ToString()` is both relatively fast (string comparisons are actually fairly fast internally?) and fairly reasonable.  Another option would be `GetHashCode()`

You might get slightly faster performance using a `HashSet` rather than a `List`, in which case....you don't care (don't need to check), because if you add the same thing to a HashSet twice...it's only in there once!

Reply:
OK. I expected it to be worse.